src/gtk/toga_gtk/widgets/internal/buttons/refresh.py

```diff
@@ -0,0 +1,190 @@
+from toga_gtk.libs import Gtk
+from .base import ParentPosition
+
+
+class RefreshButtonWidget(Gtk.Revealer):
+    def __init__(self, on_refresh: callable, on_close: callable, 
+                 position: Gtk.Align, margin: int, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.parent = None
+
+        self._refresh_btn = Gtk.Button.new_from_icon_name(
+            "view-refresh-symbolic", Gtk.IconSize.BUTTON)
+        self._refresh_btn.set_can_focus(False)
+        refresh_btn_context = self._refresh_btn.get_style_context()
+        refresh_btn_context.add_class("osd")
+        refresh_btn_context.add_class("toga-detailed-list-floating-buttons")
+
+        self._refresh_btn_handler = self._refresh_btn.connect(
+            "clicked",
+            lambda w: on_refresh())
```

Comment:
As before - we're in control of the handler, so we can give it whatever prototype we want. In addition, since it's a GTK handler, we should use the `gtk_` prefix.

It's possible that this might also fall foul of binding the widget to a specific handler; the handlers can be changed at runtime; we should be making a permanent event binding to an internal method, and using *that* method to direct to the "current" `on_refresh()` handler, rather than binding directly to the `on_refresh` that was in place when the button was created.

Reply:
Added `gtk_` here and to the class below.

That is an internal method. The class below `RefreshButton` has a `set_on_refresh` method. This class `RefreshButtonWidget` only handles the refresh button presentation, since there will be two buttons it makes sense to move the code to its own class.